AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -23,6 +23,10 @@ module {:options "/functionSyntax:4" } Mutations {
   import MutationErrorRefinement
   import KmsUtils
 
+  datatype InitialHVVerificationState =
+    | TerminalHV1()
+    | TerminalHV2(decryptedBranchKey: KMS.DecryptResponse)
```

Comment:
Names are tough. 
If you think the name is not clear, you can always add a comment.

Though now that I think about it,
we need to call decrypt if any of the follow are true:
- strategy is decrypt/encrypt
- terminal is HV-2
- original is HV-2

I am increasingly think mutations are too damn complicated... 
if I had known that we were going to do HV-2 a year ago,
I would never have had any option by decrypt... 

For M2, let's focus on terminal HV-2;
that is what our customers need.
cc: @josecorella 

Actually... here... we can do this.
```suggestion
  datatype ActiveVerificationHolder =
    | NotDecrypt()
    | KmsDecrypt(kmsRes: KMS.DecryptResponse)
```